src/ui/components/NewAccordion/Accordion.tsx

```diff
@@ -0,0 +1,162 @@
+import classNames from "classnames";
+import React, { MouseEventHandler, useReducer } from "react";
+import {
+  collapseSection,
+  endResizing,
+  expandSection,
+  getInitialState,
+  getIsCollapsed,
+  getIsIndexResizable,
+  getIsResizing,
+  getPosition,
+  getResizingParams,
+  reducer,
+  resize,
+  startResizing,
+} from "./reducer";
+import { BORDER_HEIGHT, HANDLE_HEIGHT } from "./utils";
+
+export interface AccordionItem {
+  className?: string;
+  component: React.ReactNode;
+  header: string;
+  onToggle?: (open: boolean) => void;
+  collapsed?: boolean;
+  button?: React.ReactNode;
+}
+export interface SectionPosition {
+  top: number;
+  height: number;
+}
+export type CollapsedState = boolean[];
+export type CreasesState = number[];
+
+function ResizeHandle({
+  onResizeStart,
+  isResizing,
+}: {
+  onResizeStart: (e: React.MouseEvent) => void;
+  isResizing: boolean;
+}) {
+  return (
+    <div
+      className={classNames(
+        "absolute w-full border-transparent hover:border-blue-400",
+        isResizing ? "border-blue-400" : ""
+      )}
+      style={{
+        cursor: "ns-resize",
+        borderBottomWidth: `${HANDLE_HEIGHT}px`,
+        top: `${BORDER_HEIGHT - HANDLE_HEIGHT}px`,
+      }}
+      onMouseDown={onResizeStart}
+    />
+  );
+}
+
+function Section({
+  children,
+  index,
+  isCollapsed,
+  isBeingResized,
+  isResizable,
+  isResizing,
+  position,
+  toggleCollapsed,
+  onResizeStart,
+}: {
+  children: React.ReactNode;
+  index: number;
+  isBeingResized: boolean;
+  isCollapsed: boolean;
+  isResizable: boolean;
+  isResizing: boolean;
+  position: SectionPosition;
+  toggleCollapsed: (index: number) => void;
+  onResizeStart: (e: React.MouseEvent) => void;
+}) {
+  return (
+    <div
+      className="absolute w-full h-full"
+      style={{ ...position, transition: isResizing ? "" : "all 0.1s ease-in-out" }}
+    >
+      {isResizable && <ResizeHandle onResizeStart={onResizeStart} isResizing={isBeingResized} />}
+      <div className="overflow-hidden flex flex-col w-full h-full">
+        <div className={classNames("border-b", index > 0 ? "border-black" : "border-gray-200")} />
+        <button
+          className="font-bold flex space-x-2 bg-gray-200 w-full px-2"
```

Comment:
This is fine for now, but similar to the API question, I wonder what we'd need to do for this to be headless and more extendable.